packages/near-api-js/src/account.ts

```diff
@@ -710,4 +711,100 @@ export class Account {
             total: summary.total.toString(),
         };
     }
+
+    createTransaction(recevier: Account | string): Transaction {
+        return new Transaction(this, recevier);
+    }
+  
+    async hasDeployedContract(): Promise<boolean> {
+        return (await this.state()).code_hash !== EMPTY_CONTRACT_HASH;
+    }
 }
+
+
+/**
+ * Transaction Builder class. Initialized to an account that will sign the final transaction
+ */
+export class Transaction {
+    readonly receiverId: string;
+    readonly actions: Action[] = [];
+    private accountToBeCreated = false;
+    private _transferAmount?: BN;
+
+    constructor(private signer: Account, receiver: Account | string) {
+        this.receiverId = typeof receiver === 'string' ? receiver : receiver.accountId;
+    }
+
+    addKey(publicKey: string | PublicKey, accessKey: AccessKey = fullAccessKey()): this {
+        this.actions.push(addKey(PublicKey.from(publicKey), accessKey));
+        return this;
+    }
+
+    createAccount(): this {
+        this.accountToBeCreated = true;
+        this.actions.push(createAccount());
+        return this;
+    }
+
+    deleteAccount(beneficiaryId: string): this {
+        this.actions.push(deleteAccount(beneficiaryId));
+        return this;
+    }
+
+    deleteKey(publicKey: string | PublicKey): this {
+        this.actions.push(deleteKey(PublicKey.from(publicKey)));
+        return this;
+    }
+
+    deployContract(code: Uint8Array | Buffer): this {
+        this.actions.push(deployContract(code));
+        return this;
+    }
+
+    functionCall(
+        methodName: string,
+        args: Record<string, unknown> | Uint8Array,
+        {
+            gas = DEFAULT_FUNCTION_CALL_GAS,
+            attachedDeposit = ZERO_NEAR,
+        }: { gas?: BN; attachedDeposit?: BN } = {},
+    ): this {
+        this.actions.push(
+            functionCall(methodName, args, gas, attachedDeposit),
+        );
+        return this;
+    }
+
+    stake(amount: BN, publicKey: PublicKey | string): this {
+        this.actions.push(stake(amount, PublicKey.from(publicKey)));
+        return this;
+    }
+
+    transfer(amount: BN): this {
+        this._transferAmount = amount;
+        this.actions.push(transfer(this._transferAmount));
+        return this;
+    }
+
+    push(action: Action): this {
+        this.actions.push(action);
+        return this;
+    }
+
+    get accountCreated(): boolean {
```

Comment:
What is the use case for this (and `transferAmount`)? Without more context it seems like it would be better for the calling code to manage these values since they don't seem related to building a set of transactions.

Reply:
This class originally lived with workspaces-js and it needed to track whether an account was created, because the testnet workspace would delete these accounts refunding the root account after a test. It should instead extend this class.

Reply:
Agreed I don't think this class needs to maintain this metadata about the transaction. Also seems to me that these can be derived from the action list. Removing